### PR TITLE
Fix server script status

### PR DESCRIPTION
**Pull request by dback2**
Fix server script status not getting the callback to show the actual status texts instead of "[]". Show "[ Fetching status ] " on refresh calls instead of "[" + existing text + "]".

Fixes: https://highfidelity.fogbugz.com/f/cases/16876/

Test Plan:
- Enter Interface in desktop mode in your sandbox (not serverless)
- Find an entity with a server script on it or import the following entity:
https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrower.json
- Open Create and view the Properties tab for the above entity
- Verify next to Server Script Status that the text shows as "Running"
- Press the reload button next to Server Script URL field
- Verify the status text first appears as "[ Fetching status ] " before possibly changing to "Loading" (this status might get quickly skipped) and finally showing as "Running"
- Mash the reload button many times 
- Verify the status text still goes from "[ Fetching status ] " to ending with "Running" and nothing else unexpected is displayed
- Delete the text in the Server Script URL field
- Verify once clicking out of the box that the status text appears as "[ Fetching status ] " before changing to "Not running"
- Enter "a" or any other non-URL in the Server Script URL field
- Verify once clicking out of the box that the status text first appears as "[ Fetching status ] " before changing to "Error loading script" with error message in the box below it
- Put the original test back into the Server Script URL field which was:
https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrowerServer.js
- Verify once clicking out of the box that the status text first appears as "[ Fetching status ] " before ending with "Running"

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11888/HighFidelity-Beta-PR13765-e5abd9d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11888/HighFidelity-Beta-PR13765-e5abd9d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11888/HighFidelity-Beta-Interface-PR13765-e5abd9d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11888/HighFidelity-Beta-PR13765-e5abd9d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11888/HighFidelity-Beta-Interface-PR13765-e5abd9d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11891/HighFidelity-Beta-PR13765-f9e25d3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11891/HighFidelity-Beta-PR13765-f9e25d3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11891/HighFidelity-Beta-Interface-PR13765-f9e25d3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11891/HighFidelity-Beta-PR13765-f9e25d3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11891/HighFidelity-Beta-Interface-PR13765-f9e25d3.exe

**Comment by thoys** [comment]
Needs testplan @dback2

File: scripts/system/edit.js [hunk 1]
```diff
@@ -2038,10 +2038,15 @@ var PropertiesTool = function (opts) {
     };
 
     that.setVisible(false);
+    
+    function emitScriptEvent(data) {
+        webView.emitScriptEvent(JSON.stringify(data));
+        createToolsWindow.emitScriptEvent(JSON.stringify(data));
```
Comment (huffman) [inline scripts/system/edit.js]: Could we instead call `JSON.stringify` once?

File: scripts/system/html/js/entityProperties.js [hunk 2]
```diff
@@ -36,6 +36,8 @@ var lastEntityID = null;
 
 var MATERIAL_PREFIX_STRING = "mat::";
 
+var INVALIDATE_SCRIPT_STATUS = "[ Fetching status ]";
```
Comment (huffman) [inline scripts/system/html/js/entityProperties.js]: Should this be `INVALID_SCRIPT_STATUS`, or `PENDING_SCRIPT_STATUS`?

File: scripts/system/edit.js (hunk 1 above)
Comment (dback2) [reply scripts/system/edit.js]: Good point

File: scripts/system/html/js/entityProperties.js (hunk 2 above)
Comment (dback2) [reply scripts/system/html/js/entityProperties.js]: Yeah I was debating the name, agreed PENDING_SCRIPT_STATUS is probably better.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11910/HighFidelity-Beta-PR13765-d886152.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11910/HighFidelity-Beta-PR13765-d886152.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11910/HighFidelity-Beta-Interface-PR13765-d886152.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11910/HighFidelity-Beta-PR13765-d886152.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11910/HighFidelity-Beta-Interface-PR13765-d886152.exe

**Comment by qlistephens95** [comment]
Currently Testing

**Comment by qlistephens95** [comment]
FAIL @conklin94122 . Server script status shows "failed to retrieve status" instead of "running".
Steps to reproduce:
1: Launch assumed build.
2: import entity https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrower.json
3: Go to properties tab of imported entity.
4: Go to server script status at the bottom of properties.
5: Server script status shows "failed to retrieve status" instead of "running".
Tested on Mac/desk and Win/desk.
Does not happen on Master or Stable.
Screenshot attached
screen shot 2018-08-09 at 10 54 26 am

**Comment by dback2** [comment]
@qlistephens95 please re-test in your sandbox instead of serverless sorry I should have specified

**Comment by qlistephens95** [comment]
Currently Testing

**Comment by qlistephens95** [comment]
FAIL @conklin94122 server script status shows "error loading script" when cutting server script URL text and re-inputting https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrower.json.
(BEFORE DELETING AND RE ADDING SCRIPT) screen shot 2018-08-09 at 2 49 02 pm
(AFTER DELETING AND RE ADDING SCRIPT) screen shot 2018-08-09 at 2 49 34 pm
Steps to reproduce:
1: Launch assumed build.
2: import entity https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrower.json
3: Go to properties tab of imported entity.
4: Go to server script status at the bottom of properties.
5: Delete Server Script and re add URL listed above
6: Instead of getting "running" message you get "error loading content" message.
Tested on Mac/desk and Win/desk.
Does not happen on Master or Stable.

**Comment by dback2** [comment]
@qlistephens95 This is a good test case that I forgot to add to the test plan nice find. The URL for https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrower.json should only be used to import the entity. When replacing the URL in the Server Script URL field you should copy the one that is existing in that field whose path ends in .js.

**Comment by qlistephens95** [comment]
PASS @conklin94122 verified that when importing https://hifi-content.s3.amazonaws.com/davidback/production/flamethrower/flamethrower.json and going to properties and deleting the server script URL, server script shows as "not running". When Re adding the server script URL that was already there "not running" changes to "running". Everything works as expected. Sorry for the mishap.

**Pull request merged**